.github/actions/pnpm-store-cache/action.yml

```diff
@@ -0,0 +1,20 @@
+name: Cache PNPM store
+description: Caches the PNPM store to speed up the build.
+
+runs:
+  using: composite
+  steps:
+    - id: weekly-cache-key
+      name: Key for weekly rotation of cache
+      shell: bash
+      run: echo "key=pnpm-store-`date -u "+%Y-%U"`" >> $GITHUB_OUTPUT
+
+    - uses: actions/cache@v4
```

Comment:
Pretty sure this will result in the cache being updated constantly; this should use actions/cache when the cache is empty, otherwise it should use actions/cache/restore.